core/trino-main/src/main/java/io/trino/execution/SetColumnTypeTask.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.trino.Session;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.metadata.RedirectionAwareTableHandle;
+import io.trino.metadata.TableHandle;
+import io.trino.security.AccessControl;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.TypeNotFoundException;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.SetColumnType;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.util.concurrent.Futures.immediateVoidFuture;
+import static io.trino.metadata.MetadataUtil.createQualifiedObjectName;
+import static io.trino.spi.StandardErrorCode.COLUMN_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.COLUMN_TYPE_UNKNOWN;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.TYPE_NOT_FOUND;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static io.trino.sql.analyzer.TypeSignatureTranslator.toTypeSignature;
+import static io.trino.type.UnknownType.UNKNOWN;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class SetColumnTypeTask
+        implements DataDefinitionTask<SetColumnType>
+{
+    private final Metadata metadata;
+    private final TypeManager typeManager;
+    private final AccessControl accessControl;
+
+    @Inject
+    public SetColumnTypeTask(Metadata metadata, TypeManager typeManager, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "ALTER COLUMN";
```

Comment:
`SET COLUM TYPE` ?

Reply:
`ALTER COLUMN` is the same layer as the existing task names. e.g. `ADD COLUMN` & `DROP COLUMN`.
```
<alter table action> ::=
    <add column definition> 
  | <alter column definition> 
  | <drop column definition> 
  | <add table constraint definition> 
  | <alter table constraint definition> 
  | <drop table constraint definition> 
  | <add table period definition> 
  | <drop table period definition> 
  | <add system versioning clause> 
  | <drop system versioning clause> 
```

Reply:
> `ALTER COLUMN` is the same layer as the existing task names. e.g. `ADD COLUMN` & `DROP COLUMN`.

I'm not sure I follow your explanation.

Reply:
It should be "SET COLUMN TYPE". This is an informational name for reporting purposes and does not need to related to the SQL syntax.

Reply:
Changed to "SET COLUMN TYPE".